src/typeInference/annotateCastOperators.ts

```diff
@@ -0,0 +1,24 @@
+import {
+	SequenceMultiplicity,
+	SequenceType,
+	stringToValueType,
+	ValueType,
+} from '../expressions/dataTypes/Value';
+import { IAST } from '../parsing/astHelper';
+import { insertAttribute } from './insertAttribute';
+
+export function annotateCastOperator(ast: IAST): SequenceType | undefined {
+	// when ast is not annotated, ast[2] goes to the singleType, ast[2][1] the atomicType
+	// ast[2][1][1]['prefix'] the prefix 'xs' and then ast[2][1][2] is the targetType
+	const targetTypeFromAST = ast[2][1][1]['prefix'] + ':' + ast[2][1][2];
```

Comment:
Could this potentially be replaced by a chain of getAttribute or getChild calls?

Reply:
Fixed in commit 81eee278900d0dd312205ee59adc3b278befb07c